util/perl/OpenSSL/paramnames.pm

```diff
@@ -0,0 +1,663 @@
+    'LIBSSL_RECORD_LAYER_PARAM_MODE' =>           "mode",
+    'LIBSSL_RECORD_LAYER_PARAM_READ_AHEAD' =>     "read_ahead",
+    'LIBSSL_RECORD_LAYER_READ_BUFFER_LEN' =>      "read_buffer_len",
+    'LIBSSL_RECORD_LAYER_PARAM_USE_ETM' =>        "use_etm",
+    'LIBSSL_RECORD_LAYER_PARAM_STREAM_MAC' =>     "stream_mac",
+    'LIBSSL_RECORD_LAYER_PARAM_TLSTREE' =>        "tlstree",
+    'LIBSSL_RECORD_LAYER_PARAM_MAX_FRAG_LEN' =>   "max_frag_len",
+    'LIBSSL_RECORD_LAYER_PARAM_MAX_EARLY_DATA' => "max_early_data",
+    'LIBSSL_RECORD_LAYER_PARAM_BLOCK_PADDING' =>  "block_padding",
+);
+
+# Generate string based macros for public consumption
+sub generate_public_macros {
+    my @macros = ();
+
+    foreach my $name (keys %params) {
+        my $val = $params{$name};
+        my $def = '# define OSSL_' . $name . ' ';
+
+        if (substr($val, 0, 1) eq '*') {
+            $def .= 'OSSL_' . substr($val, 1);
+        } else {
+            $def .= '"' . $val . '"';
+        }
+        push(@macros, $def)
+    }
+    return join("\n", sort @macros);
+}
+
+# Generate number based macros for internal use
+# The numbers are unique per string
+sub generate_internal_macros {
+    my @macros = ();
+    my $count = 0;
+    my %reverse;
+
+    # Determine the number for each unique string
+    # Sort the names to improve the change of cache coherency
+    foreach my $name (sort keys %params) {
+        my $val = $params{$name};
+
+        if (substr($val, 0, 1) ne '*' and not defined $reverse{$val}) {
+            $reverse{$val} = $count++;
+        }
+    }
+
+    # Output the defines
+    foreach my $name (keys %params) {
+        my $val = $params{$name};
+        my $def = '#define PIDX_' . $name . ' ';
+
+        if (substr($val, 0, 1) eq '*') {
+            $def .= 'PIDX_' . substr($val, 1);
+        } else {
+            $def .= $reverse{$val};
+        }
+        push(@macros, $def)
+    }
+    return "#define NUM_PIDX $count\n\n" . join("\n", sort @macros);
+}
+
+sub generate_trie {
+    my %trie;
+    my $nodes = 0;
+    my $chars = 0;
+
+    foreach my $name (keys %params) {
+        my $val = $params{$name};
+        if (substr($val, 0, 1) ne '*') {
+            my $cursor = \%trie;
+
+            $chars += length($val);
+            for my $i (0 .. length($val) - 1) {
+                my $c = substr($val, $i, 1);
+
+                if (not $case_sensitive) {
+                    $c = '_' if $c eq '-';
+                    $c = lc $c;
+                }
+
+                if (not defined $$cursor{$c}) {
+                    $$cursor{$c} = {};
+                    $nodes++;
+                }
+                $cursor = %$cursor{$c};
+            }
+            $$cursor{'val'} = $name;
+        }
+    }
+    #print "\n\n/* $nodes nodes for $chars letters*/\n\n";
+    return %trie;
+}
+
+sub generate_code_from_trie {
+    my $n = shift;
+    my $trieref = shift;
+    my $indent = 1 * $n + 1;
```

Comment:
IMO, I think the indent should be as per our code standards. Don't worry about long lines.

Reply:
The generated code is almost totally unreadable to a human, I just don't see the point of _proper_ indentation.